### PR TITLE
ExpenseTrackrr

### PR DESCRIPTION
**Pull request by codepantherr**
## Expense Trackrr has the following 👇 

✅ Followed gitflow

A splash page.
A login page
A registration page.
A category page as home page
A form to create new categories
A management page to see all your purchase in a specific category
A form to add new purchases to your app in the current category and even you can select others.
App deployed using Heroku [Live demo](https://expensetrackrr.herokuapp.com/)
Video presentation [here](https://www.loom.com/share/bbf92d01c2da4b6fb925ba5905076bb1?sharedAppSource=personal_library)

### CODE REVIEW

File: spec/helpers/group_helper_spec.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+# Specs in this file have access to a helper object that includes
+# the GroupHelper. For example:
+#
+# describe GroupHelper do
+#   describe "string concat" do
+#     it "concats two strings with spaces" do
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
```
Comment (Emmyn5600) [inline spec/helpers/group_helper_spec.rb:12]: - [ ] Kindly, implement the pending tests or you can remove the commented ones to keep the running tests only.

image

File: test/models/user_test.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class UserTest < ActiveSupport::TestCase
+  # test "the truth" do
+  #   assert true
+  # end
```
Comment (Emmyn5600) [inline test/models/user_test.rb:6]: - [ ] Remove the commented code.

File: spec/helpers/expenses_helper_spec.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+# Specs in this file have access to a helper object that includes
+# the ExpensesHelper. For example:
+#
+# describe ExpensesHelper do
+#   describe "string concat" do
+#     it "concats two strings with spaces" do
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
```
Comment (Emmyn5600) [inline spec/helpers/expenses_helper_spec.rb:12]: - [ ] Kindly, implement the pending tests or you can remove the commented ones to keep the running tests only.

image

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @codepantherr 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight :champagne:

- All linters are passing :heavy_check_mark:
- Good commit history :heavy_check_mark:
- Good PR description :heavy_check_mark:
- The database is well structured/created. :heavy_check_mark:
- You have implemented almost all the requirements. :heavy_check_mark:
- Some of the tests are passing successfully:heavy_check_mark:

## Required Changes :recycle:

- [ ] Check the comment under the review

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!:clap::clap::clap:

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/helpers/group_helper_spec.rb (hunk 1 above)
Comment (codepantherr) [reply spec/helpers/group_helper_spec.rb]: Hi @Emmyn5600 👋 This is resolved.

File: test/models/user_test.rb (hunk 2 above)
Comment (codepantherr) [reply test/models/user_test.rb]: This has been resolved 💯

File: spec/helpers/expenses_helper_spec.rb (hunk 3 above)
Comment (codepantherr) [reply spec/helpers/expenses_helper_spec.rb]: This has been fixed

**Review by enaburekhan — APPROVED** [review]
Hi Codepantherr, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Code review by **Eric Enaburekhan**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**